programs/drift/src/instructions/admin.rs

```diff
@@ -4845,16 +4843,103 @@ pub fn handle_update_if_rebalance_config(
     Ok(())
 }
 
-pub fn handle_zero_amm_fields_prep_mm_oracle_info(ctx: Context<UpdateAmmParams>) -> Result<()> {
-    let mut perp_market = ctx.accounts.perp_market.load_mut()?;
-    perp_market.amm.long_intensity_count = 0;
-    perp_market.amm.short_intensity_count = 0;
-    perp_market.amm.max_position_size = 0;
-    perp_market.amm.total_fee_earned_per_lp = 0;
-    msg!(
-        "zeroed amm fields for perp market {}",
-        perp_market.market_index
+pub fn handle_update_mm_oracle(
+    ctx: Context<Empty>,
+    oracle_price: i64,
+    oracle_sequence_id: u64,
+) -> Result<()> {
+    let remaining_accounts = &ctx.remaining_accounts;
+
+    let signer = &remaining_accounts[1];
+
+    // Verify this ix is allowed
+    let state = &remaining_accounts[2].data.borrow();
+    assert!(state[982] & 1 == 0);
```

Comment:
how much extra CUs does this add btw? I think you might be able to toggle mm oracle off by setting value to 0. When admin enables, set the mm oracle to current oracle. Then you dont need to load this